### ISSUE TITLE
tab completion error

### ISSUE THREAD

**Issue by pyeguy** [xonsh/xonsh#2554]
hi all

i'm on windows 10 using `cmder` (essentially ConEmu) sometimes i get the following error when tab completing directories:

```python
 cd monomersException in thread Thread-163:
Traceback (most recent call last):
  File "C:\Users\camer\Anaconda3\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\Users\camer\Anaconda3\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\camer\Anaconda3\lib\site-packages\prompt_toolkit\interface.py", line 860, in run
    completions = list(buffer.completer.get_completions(document, complete_event))
  File "C:\Users\camer\Anaconda3\lib\site-packages\xonsh\ptk\completer.py", line 49, in get_completions
    self.ctx)
  File "C:\Users\camer\Anaconda3\lib\site-packages\xonsh\__amalgam__.py", line 124, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "C:\Users\camer\Anaconda3\lib\site-packages\xonsh\completers\__amalgam__.py", line 1467, in complete_cd
    return complete_dir(prefix, line, start, end, ctx, True)
  File "C:\Users\camer\Anaconda3\lib\site-packages\xonsh\completers\__amalgam__.py", line 1201, in complete_dir
    filtfunc=os.path.isdir)
  File "C:\Users\camer\Anaconda3\lib\site-packages\xonsh\completers\__amalgam__.py", line 1159, in complete_path
    sort_result=glob_sorted):
  File "C:\Users\camer\Anaconda3\lib\site-packages\xonsh\__amalgam__.py", line 7052, in iglobpath
    return _iglobpath(s, ignore_case=ignore_case, sort_result=sort_result)[0]
  File "C:\Users\camer\Anaconda3\lib\site-packages\xonsh\__amalgam__.py", line 7034, in _iglobpath
    paths = glob.glob(s, recursive=True)
  File "C:\Users\camer\Anaconda3\lib\glob.py", line 20, in glob
    return list(iglob(pathname, recursive=recursive))
  File "C:\Users\camer\Anaconda3\lib\glob.py", line 55, in _iglob
    yield from _glob1(dirname, basename, dironly)
  File "C:\Users\camer\Anaconda3\lib\glob.py", line 83, in _glob1
    return fnmatch.filter(names, pattern)
  File "C:\Users\camer\Anaconda3\lib\fnmatch.py", line 59, in filter
    for name in names:
  File "C:\Users\camer\Anaconda3\lib\glob.py", line 82, in <genexpr>
    names = (x for x in names if not _ishidden(x))
  File "C:\Users\camer\Anaconda3\lib\glob.py", line 153, in _ishidden
    return path[0] in ('.', b'.'[0])
IndexError: string index out of range
```

i can't quite isolate the behavior that causes it but it seems like it's when i'm doing `cd <partial_dir_name>` + `tab` and the partial dir name has history autocomplete suggestion that isn't in the local directory. 

for instance if previously i've done `cd thing` and i'm in a directory that has a folder called `thingy` that I want to tab complete from `cd  thin` xonsh will have the historical `cd thing` that i could select with an `arrow up` but when i hit `tab` to get the local tab completion result instead i get the error. 
I assume this is more of a `prompt toolkit` issue?

**Comment by scopatz** [comment]
Thanks for reporting @pyeguy. This probably isn't a tab completion issues,  but is likely related to invalid input going to glob.... It would be nice to figure out what this is... Any more details that you can provide, would be most welcome!

**Comment by pyeguy** [comment]
it's pretty strange, it's actually just directory dependent. if i'm in one particular dir and i attempt any tab completion ( even `tab` with nothing typed in the prompt) i get the glob error...

I should probably mention that this is in a[ Google Drive File Stream](https://gsuiteupdates.googleblog.com/2017/09/drive-file-stream-from-google.html) folder which could add some funkiness but all the parent directories work fine...

**Comment by scopatz** [comment]
Ahh interesting.  Maybe it can't glob correctly because of that.  It is as if an empty string is being returned for some reason by `glob._iterdir()`.  In that directory, in normal python can you try the following?

```console
$ python -c "import glob; glob.glob('/path/to/dir/*')"
```

This should produce a similar error message.

**Comment by scopatz** [comment]
Hi @pyeguy - I have a potential fix in #2555. That said, I think that this is a potential bug in Python.

**Comment by gforsyth** [comment]
Fixed in #2555
